lib/Builtins.cpp

```diff
@@ -514,7 +514,7 @@ std::string Builtins::GetMangledTypeName(Type *Ty) {
       std::string AS_name = "AS" + std::to_string(AS);
       mangled_type_str += "U" + std::to_string(AS_name.size()) + AS_name;
     }
-    if (!Ty->isOpaquePointerTy()) {
+    if (!Ty->isPointerTy()) {
```

Comment:
This isn't correct. The if should just be removed. Debug builds would assert if called on a non-pointer or opaque pointer.